### PR TITLE
Redesign of Prow deployment

### PR DESCRIPTION
**Pull request by gabrielecerami**
This is a series of patches to add a redesign of the prow deployment, to lower maintenance and add tests.

### CODE REVIEW

**Comment by gabrielecerami** [comment]
/cc @dhiller 
/cc @rmohr 
Github identified the token in my code and revoked it. I'm exploring how to pass secret to jobs in prow.

**Comment by rmohr** [comment]
@danielBelenky I think you PR does not rebase, can that be? It sees all these jobs as added. In fact I removed all these jobs and the only exist on the PR of @gabrielecerami since it is based on an older state. @gabrielecerami I think if you rebase that awkwardness would  disappear. @danielBelenky something which we should fix.

File: github/ci/prow-deploy/README.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+## How to launch a deployment
+
+First you need to create a deployment configuration file.
+The role will look by default in
+
+    /etc/prow-deploy/deploy_config.yaml
+    
+An example can be found in directory
+
+    molecule/default/vars/deploy_config.yaml
```
Comment (fgimenez) [inline github/ci/prow-deploy/README.md:53]: I can't find this file, is it added as part of this PR?

**Comment by fgimenez** [comment]
/lgtm

All ok in the last build with the current status https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_project-infra/608/prow-deploy-test/1334035927168716800 

@rmohr this is ready to go PTAL.

**Comment by rmohr** [comment]
/lgtm
/approve

@gabrielecerami @fgimenez thank you for you hard work. :+1:

**Comment by rmohr** [comment]
/hold

Let's see if wa can sort out the owner file issue

**Pull request merged**